### ISSUE TITLE
No indentation after metadata form

### ISSUE THREAD

**Issue by laurentpetit** [laurentpetit/ccw#602]
Original [issue 561](https://code.google.com/p/counterclockwise/issues/detail?id=561) created by laurentpetit on 2013-04-05T12:53:01.000Z:

Pressing TAB on line after metadata form does not increase indent (any indentation characters are removed from beginning of the line). 
This happens no matter what indentation previous line has. 
For example in following code

(ns ^{:doc &quot;&quot;}
something)

Pressing tab on second line removes any indentation. 
CCW version 0.13.0.201304042121

**Comment by gofeddy** [comment]
This would be really beneficial to have since we make use of CCW's auto-indenting capabilities on Eclipse. Every time we write a `deftest` with a `^{:doc ..}` metadata form, the auto-indentation feature breaks the indentation on the next line as mentioned in the comment above and we have to manually indent these lines all over again. 

Having this feature would help avoid this minor issue.

Just checking, has this been prioritized to be on a released version of CCW any time soon?

**Comment by laurentpetit** [comment]
Sure. We can plan it either for next stable or the one following it. 

By the way, what would be a correct indentation?

- align the object with its metadata?
- indent the object with the parent form as if there were no metadata ?

A couple examples that could serve as regression tests would be appreciated

**Comment by gofeddy** [comment]
I'm in favor of option 2 i.e indent the object with the parent form as if there were no metadata.

My thoughts were that since metadata just provides additional description/detail/accessory to the actual data and is not the "real data" itself, from an indentation point of view, it made sense to me to treat it as though there were no metadata. 

Ex:

Before:
```
(deftest ^{:doc "Some doc
                 More doc."}
<test-name>
<body>)
```
After:
```
(deftest ^{:doc "Some doc
                 More doc."}
  <test-name>
  <body>)
```

Of course, this is just my personal preference, others might have their opinions too, which can be perfectly valid for their use cases.

**Comment by arichiardi** [comment]
BTW, Isn't there any rule [here](https://github.com/bbatsov/clojure-style-guide) ?

**Comment by laurentpetit** [comment]
@arichiardi I browsed the link, and didn't see any.

**Comment by arichiardi** [comment]
Yes neither did I....I am considering opening an issue...if you both don't mind...

**Comment by laurentpetit** [comment]
An issue where, about what?



—
Envoyé avec Mailbox

**Comment by arichiardi** [comment]
Sorry Laurent for the short message...I meant to open a issue on ```clojure-style-guide``` and see what other devs have to say about this case..

**Comment by laurentpetit** [comment]
Why not. To be honest, I don't know the popularity of clojure-style-guide. 



—
Envoyé avec Mailbox

**Comment by arichiardi** [comment]
I like it a lot :+1:

**Comment by laurentpetit** [comment]
long-standing bug, not striclty necessary for `0.32.0`. Moving it to `next ` milestone

**Comment by gofeddy** [comment]
@laurentpetit, I see that 0.32.0 has already been released and looks like this was pushed out to a future milestone. Do you happen to have an estimated timeline for this? Thanks!
